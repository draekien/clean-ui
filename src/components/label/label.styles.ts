import { ThemeUIStyleObject } from '@theme-ui/css';

export const labelCss: ThemeUIStyleObject = {
  variant: 'text.eyebrow',
  color: 'text-muted',
};
