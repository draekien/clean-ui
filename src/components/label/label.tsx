/** @jsxImportSource theme-ui */
import React from 'react';
import { Text } from '../..';
import { colors } from '../theme/theme.colors';

export interface LabelProps
  extends Omit<React.HTMLAttributes<HTMLLabelElement>, 'color'> {
  htmlFor: string;
  color?: keyof typeof colors;
}

export const Label: React.FC<LabelProps> = ({ color = 'text-muted', ...rest }) => {
  return (
    <label {...rest}>
      <Text variant="eyebrow" color={color}>
        {rest.children}
      </Text>
    </label>
  );
};
