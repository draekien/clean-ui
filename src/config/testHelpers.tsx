import React, { ReactNode } from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from '../components/theme/theme.provider';

export const themedRender = (element: ReactNode) =>
  render(<ThemeProvider>{element}</ThemeProvider>);
