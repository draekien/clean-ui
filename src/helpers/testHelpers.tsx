import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from '../components/theme/theme.provider';

export const themedRender = (element: React.ReactNode) => {
  return render(<ThemeProvider>{element}</ThemeProvider>);
};

export const hexToRgb = (hex: string) => {
  hex = hex.replace(/[^0-9A-F]/gi, '');
  const bigint = parseInt(hex, 16);
  const r = (bigint >> 16) & 255;
  const g = (bigint >> 8) & 255;
  const b = bigint & 255;

  return `rgb(${r}, ${g}, ${b})`;
};
