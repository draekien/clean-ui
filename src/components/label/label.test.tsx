import React from 'react';
import { themedRender } from '../../helpers/testHelpers';
import { Label } from './label';

describe('Label component', () => {
  it('should match snapshot', () => {
    const { container } = themedRender(<Label htmlFor="myId">My label</Label>);

    expect(container).toMatchSnapshot();
  });
});
