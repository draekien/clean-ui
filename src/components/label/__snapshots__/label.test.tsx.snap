// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label component should match snapshot 1`] = `
<div>
  <label
    class="css-1pst1pz-Label"
    for="myId"
  >
    My label
  </label>
</div>
`;
